### PR TITLE
Specify sourceRoot rather than currentFile

### LINKED ISSUE #10: Specify `sourceRoot` rather than `currentFile` when registering breakpoint
The `sourceRoot` is the root from which all relative paths will be searched. Relative paths will not be allowed to have `..` or `.` in them.

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #10

### CODE REVIEW

File: stackdriver_debugger.c [hunk 1]
```diff
@@ -325,11 +317,15 @@ PHP_FUNCTION(stackdriver_debugger_add_snapshot)
         }
     }
 
-    if (current_file == NULL) {
-        current_file = EX(prev_execute_data)->func->op_array.filename;
-    }
+    if (source_root == NULL) {
+        source_root = zend_string_dup(EX(prev_execute_data)->func->op_array.filename, 1);
```
Comment (tmatsuo) [inline stackdriver_debugger.c]: It seems like `zend_string_dup` returns the original string when it's interned. If we're sure the `op_array.filename` is always not interned, it's safe though.

Comment (chingor13) [reply stackdriver_debugger.c]: Good point, so far in testing this hasn't been the case, but we can make sure. In this case, I'll actually dup the char*

**Comment by tmatsuo** [comment]
Does it make sense to explicitly copy the string in `stackdriver_debugger_full_filename` where the destructive operation happens so that we can just use `zend_string_dup` elsewhere?

**Comment by chingor13** [comment]
The destructive action is actually on the `php_dirname` call. It modifies the internal `char*` by inserting a null terminator at the last directory separator and returns the new length of the string.

The full amount of memory is actually retained, only some methods will only see the dirname part of the string if they iterate until a null terminator.

**Comment by tmatsuo** [comment]
Oh I see

**Review by tmatsuo — APPROVED** [review]

**Pull request merged**